### PR TITLE
Use proper scope names in ScenarioHook

### PR DESCRIPTION
**Pull request by Taluu**
The scopes names were using `feature.before` and `feature.after` in `ScenarioHook`, while it should be `scenario.before` and `scenario.after`.

Can't see which tests I should update, I'll take a look (it should be within `hooks.feature` I guess ?)

### CODE REVIEW

**Comment by ciaranmcnulty** [comment]
It's kinda worrying that this change hasn't caused any difference in the test suite

**Comment by Taluu** [comment]
pretty much agree...

**Comment by ciaranmcnulty** [comment]
Can you add a test that will fail if your change is reverted?

**Comment by Taluu** [comment]
I am using scenario scope hooks without this fix and it seems to be working. I'll try to add a test that fails without this fix, but I'm not quite sure how. I'll need some time to check it up.

**Comment by everzet** [comment]
@Taluu can you add a test so I can merge it?

**Comment by Taluu** [comment]
Will do. Any hint on how shuold I test that (and make sure a failure is triggered without this fix, otherwise it could be quite pointless) ? Because there seems to be a bunch of scenarios with before / after scenario hooks and it didn't fail before...

**Comment by everzet** [comment]
## TLDR;

There are already tests that cover this case. The reason why those weren't failing is because of defensive/type-guiard programming on my side.

## Longer version

Those constants are used during creation of hooks and scopes. They are then used by the [`HookRepository`](https://github.com/Taluu/Behat/blob/fix-scenario-scope/src/Behat/Testwork/Hook/HookRepository.php#L69) to fetch callbacks appropriate for a just fired hook. Now technically, there is a bug and in case of `ScenarioHook` this guard will be ignored for firing both `Feature` and `Scenario` hooks. So why doesn't it fail? Keep reading.

If hook and scope name check out (even wrongly), we pass forward to [filtering guard](https://github.com/Taluu/Behat/blob/fix-scenario-scope/src/Behat/Testwork/Hook/HookRepository.php#L73). Filtering guard checks if the kind of hook (all hooks in case of Behat) is the filterable hook and then asks hook to filter itself based on given `Scope`. If you drill down to [RuntimeScenarioHook](https://github.com/Taluu/Behat/blob/fix-scenario-scope/src/Behat/Behat/Hook/Call/RuntimeScenarioHook.php#L33-L35), you'll see that the first thing this guy does is type-checks the scope you pass.

Basically it means that hooks have two guards in play - name-based and type-based ones. This is redundancy and an obvious design flaw. I shouldn't have checked type inside `RuntimeScenarioHook` and should've followed tell-don't-ask querying pattern. This way already existing tests would've highlighted the name-guard bug.

## This PR and implications

With two guards in place and commitment to BC, there's not much we can do. I'll still merge this PR as it should potentially provide very minor performance optimisation by avoiding unnecessary type checks.

**Comment by Taluu** [comment]
Thanks for the explanation !

**Comment by everzet** [comment]
@Taluu thanks for spotting!
